### PR TITLE
Allow to tag sent messages with same tags as message being replied to

### PR DESCRIPTION
**Pull request by larskotthoff**
This PR introduces special values to `accounts.<account>.additional_sent_tags`, namely `*`  (apply all tags that the message being replied to, if any, has), and `-<tag>` (to exclude tags from being applied). For example, the option "foo,*,-attachment" applies the tag "foo" and any tags the message being replied to has, except for "attachment".

### CODE REVIEW

File: src/db.cc [hunk 1]
```diff
@@ -327,14 +327,39 @@ namespace Astroid {
...
   }
 
-  ustring Db::add_sent_message (ustring fname, vector<ustring> additional_sent_tags) {
+  ustring Db::add_sent_message (ustring fname, vector<ustring> additional_sent_tags, ustring parent_mid) {
     LOG (info) << "db: adding sent message: " << fname;
     additional_sent_tags.insert (additional_sent_tags.end (), sent_tags.begin (), sent_tags.end ());
+
+    if (!parent_mid.empty () &&
+            find(additional_sent_tags.begin(), additional_sent_tags.end(), "*") != additional_sent_tags.end()) {
+        notmuch_message_t * parent_msg;
```
Comment (gauteh) [inline src/db.cc]: Probably better to use `on_message` here, takes care of cleaning up the parent_msg. Might also be better to instantiate a `NotmuchMessage` on `parent_msg`, there is already a `get_tags` function which is more robust. `NotmuchMessage` should be cheap objects.

**Comment by mxmehl** [comment]
Interesting approach! #317 goes into a similar direction, especially for initial messages starting a new thread.

Could the work of this PR make realisation of #317 easier?

**Comment by gauteh** [comment]
Max writes on October 21, 2018 12:53:
> Could the work of this PR make realisation of #317 easier?

It's fairly orthogonal, but I got inspired and wrote a PR for #317: 
please test #583 for that. Post further comments on #583 on that PR.

File: src/db.cc (hunk 1 above)
Comment (larskotthoff) [reply src/db.cc]: Thanks, I'll have a look at that.

**Comment by gauteh** [comment]
Lars Kotthoff writes on October 20, 2018 21:56:
> This PR introduces special values to `accounts.<account>.additional_sent_tags`, namely '*' (apply all tags that the message being replied to, if any, has), and '-<tag>' (to exclude tags from being applied). For example, the option "foo,*,-attachment" applies the tag "foo" and any tags the message being replied to has, except for "attachment".

Does the syntax also work for the general `mail.sent_tags`?

**Comment by larskotthoff** [comment]
Good idea, I'll add that.

**Comment by larskotthoff** [comment]
I've changed the code to use `NotmuchMessage`. Turns out that this syntax for the general `mail.sent_tags` was already supported because I'm doing the processing after merging in those tags.

**Comment by larskotthoff** [comment]
Thanks, done!

**Pull request closed without merging**